### PR TITLE
test(convert/style/background): add coverage for map_extent aliases and edge-case stops

### PR DESCRIPTION
**Pull request by mitsuru**
## Summary

`convert/style/background` モジュール（`crates/fulgur/src/convert/style/background.rs`）のリージョンカバレッジが最低水準（96.15%）だったため、未カバーの分岐に対してユニットテストを追加しました。

カバレッジ変化（行カバレッジ）:
- **変更前**: 96.09%（742 行中 29 行未カバー）
- **変更後**: 96.74%（798 行中 26 行未カバー）

新たにカバーされた行（元ソースの 6 行）: `266, 270, 317, 489, 516, 517`

## Changes

- `map_extent_contain_maps_to_closest_side` — `ShapeExtent::Contain` → `RadialExtent::ClosestSide` の直接テスト（line 516）。Stylo はパース時にこの値を正規化するため CSS 経由では到達不可なアームを直接検証する。

- `map_extent_cover_maps_to_farthest_corner` — `ShapeExtent::Cover` → `RadialExtent::FarthestCorner` の直接テスト（line 517）。同上の理由。

- `linear_gradient_calc_stop_position_drops_layer` — `calc(50% + 10px)` のような混合単位ストップ位置は `to_percentage()` / `to_length()` が両方 `None` を返すため、`resolve_color_stops` がレイヤーを drop することを検証（lines 266, 270）。

- `linear_gradient_single_stop_drops_layer` — `linear-gradient(red)` のシングルストップグラデーションで `out.len() < 2` ガードを通ることを検証（line 317）。

- `conic_gradient_single_stop_drops_layer` — `conic-gradient(red)` で `stops.len() < 2` ガードを通ることを検証（line 489）。

- `first_gradient_stop_count_excludes_raster_layer` — `BgImageContent::Raster` レイヤーがグラデーションストップカウントから除外されることを検証（`_ => None` アームのカバレッジ）。

意図的に残した未カバー行と理由:

| 行 | 理由 |
|---|---|
| 160, 354, 435 | 各ディスパッチが正しい型で呼び出すため到達不可な防御的ガード |
| 168, 359, 439, 443 | 非デフォルト色補間の bail-out。Stylo がパース時に正規化するため CSS 経由では到達不可 |
| 283–300, 313–314 | `GradientItem::InterpolationHint` エラーパス。Stylo の CSS パーサーは標準 CSS から InterpolationHint アイテムを生成しない |
| 46 | `ComputedUrl::Invalid` — Stylo の URL バリデーションはペイント前に完了するため通常到達しない |
| 96 | `Image::_` catch-all — CSS `background-image` から Url/Gradient 以外の Image 型は Stylo が生成しない |

## Test plan

- [x] `cargo test -p fulgur --lib`（2159 テスト全通過）
- [x] `cargo clippy -- -D warnings`（警告なし）
- [x] `cargo fmt --check`（差分なし）
- [x] カバレッジ再測定で 6 行が新たにカバーされたことを確認

## Contributor License Agreement

By opening this pull request, I confirm that:

- [x] I have read the [Contributing Guide](https://github.com/fulgur-rs/fulgur/blob/main/CONTRIBUTING.md)
- [x] I have read the [CLA](https://github.com/fulgur-rs/fulgur/blob/main/CLA.md)
      and agree to its terms.

## Related issues

自動デイリー coverage 改善タスク（scheduled run 2026-09-07）

---
🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01FXuUQUCDoKBTQfBv7oumjn

---
_Generated by [Claude Code](https://claude.ai/code/session_01FXuUQUCDoKBTQfBv7oumjn)_

## Summary by CodeRabbit

- **テスト**
  - 背景スタイル変換における CSS エイリアスの処理を検証しました。
  - 混合単位の `calc()` を含むグラデーション停止位置が破棄されることを確認しました。
  - 単一停止の線形・円錐グラデーションが破棄されることを検証しました。
  - 停止数が不足するレイヤーの破棄を検証しました。
  - ラスターレイヤーがグラデーションとして扱われないことを確認しました。
  - PDF 出力に不要なグラデーションレイヤーが生成されないことを確認しました。

### CODE REVIEW

File: crates/fulgur/src/convert/style/background.rs [hunk 1]
```diff
@@ -1205,4 +1205,113 @@ mod tests {
...
+    // percentage nor a pure length, so `to_percentage()` and `to_length()` both
+    // return `None`. The layer is dropped rather than silently misrendering.
+
+    #[test]
+    fn linear_gradient_calc_stop_position_drops_layer() {
+        // `calc(50% + 10px)` is a mixed-unit calc — neither pure pct nor pure length.
+        // resolve_color_stops returns None for the stop, dropping the layer.
+        // The element still renders as a valid PDF without the background image.
+        let pdf = render_bg("linear-gradient(red calc(50% + 10px),blue)");
+        assert_pdf(&pdf, "calc_stop_pos");
```
[the remark this answers is not in the document]
Comment (mitsuru) [reply crates/fulgur/src/convert/style/background.rs]: ご指摘ありがとうございます。おっしゃる通り、`assert_pdf` だけでは「レイヤーが落とされた」ことを検証できていませんでした。

`ea227db` にて、3 つのテスト全てに `first_gradient_stop_count(&html).is_none()` アサーションを追加しました。これにより `resolve_color_stops` が誤ってレイヤーを保持した場合にもテストが失敗するようになります。

---
_Generated by [Claude Code](https://claude.ai/code)_

[the remark this answers is not in the document]
Comment (mitsuru) [reply crates/fulgur/src/convert/style/background.rs:1259]: ご指摘の通りです。`assert_pdf` はヘッダーの存在しか確認しないため、グラデーションレイヤーが誤って保持されていてもテストが通ってしまっていました。

`ea227db` にて修正しました。既存の `first_gradient_stop_count` ヘルパーを使い、calc 混合単位・単一 stop 線形・単一 stop 円錐の 3 テスト全てに以下のアサーションを追加しています：

```rust
assert!(first_gradient_stop_count(&html).is_none(), "...");
```

これで `resolve_color_stops` のレイヤー破棄パスが実際に機能していることを Drawables レベルで確認できます。ご確認ください。

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by mitsuru** [comment]
## CI 状況メモ: WPT ジョブの失敗について

`wpt / smoke`、`wpt / css-page`、`wpt / bugs`、`wpt / opacity-visibility`、`wpt / multicol-3` が失敗していますが、これらは本 PR の変更とは無関係です。

- `main` ブランチの最新 CI run (34012304474, conclusion: **success**) でも `wpt / bugs`・`wpt / smoke`・`wpt / opacity-visibility` が同様に failure になっています
- 本 PR の変更は `crates/fulgur/src/convert/style/background.rs` の `#[cfg(test)]` ブロックのみで、WPT reftest には影響しません
- `wpt / css-page` と `wpt / multicol-3` は main では pass しているため差分がありますが、今回追加したユニットテストが HTML reftest 結果を変えることは構造上あり得ません（フレーク扱い）

`lint`・`VRT`・`WASM build check`・`Bindings type-check`・macOS test は全て pass 済みです。残りのテストジョブ (ubuntu / arm64 / windows) は現在 in_progress です。

---
_Generated by [Claude Code](https://claude.ai/code)_

File: crates/fulgur/src/convert/style/background.rs [hunk 2]
```diff
@@ -1205,4 +1205,127 @@ mod tests {
...
+                .any(|layer| matches!(&layer.content, BgImageContent::Raster { .. }))
+        });
+        assert!(
+            has_raster,
+            "expected a Raster background layer in drawables"
+        );
+        // No gradient layer must be found: the `_ => None` arm is the raster arm.
+        // Use the pre-existing helper to avoid dead gradient match arms in this patch.
+        assert!(
+            first_gradient_stop_count(html).is_none(),
```
[the remark this answers is not in the document]
Comment (mitsuru) [reply crates/fulgur/src/convert/style/background.rs]: ご指摘ありがとうございます。おっしゃる通りで、`first_gradient_stop_count` はバンドルなしで新しい Engine を生成するため、`url(dot.png)` のラスターレイヤーが変換前に破棄されてしまい、`_ => None` アームに到達しないまま `None` が返っていました。

`153945d` にて修正しました。`first_gradient_stop_count` の呼び出しをやめ、バンドル付きで生成済みの `drawables`（実際のラスターレイヤーを含む）に対して直接チェックするようにしています：

```rust
let has_any_gradient = drawables.block_styles.values().any(|block| {
    block.style.background_layers.iter().any(|layer| {
        !matches!(&layer.content, BgImageContent::Raster { .. })
    })
});
assert!(!has_any_gradient, "a raster-only background must not produce gradient layers in drawables");
```

これにより、実際のラスターレイヤーを対象にグラデーション除外ロジックを検証でき、デッドな match アームも生じません。

---
_Generated by [Claude Code](https://claude.ai/code)_

File: crates/fulgur/src/convert/style/background.rs [hunk 3]
```diff
@@ -1205,4 +1205,135 @@ mod tests {
...
+        // Verify that none of the layers in the already-built drawables (which includes
+        // the raster layer above) is a non-raster/gradient type. This exercises the
+        // raster-exclusion logic on an actual raster layer present in `drawables`.
+        let has_any_gradient = drawables.block_styles.values().any(|block| {
+            block
+                .style
+                .background_layers
+                .iter()
+                .any(|layer| !matches!(&layer.content, BgImageContent::Raster { .. }))
+        });
```
[the remark this answers is not in the document]
Comment (mitsuru) [reply crates/fulgur/src/convert/style/background.rs]: Addressed in commit `0d3dbe2`. Extracted `first_gradient_stop_count_in_drawables(drawables: &crate::drawables::Drawables) -> Option<usize>` exactly as suggested. The raster test now builds drawables with a real bundle (so the raster layer survives), then passes `&drawables` directly to the extracted helper — this exercises the `_ => None` arm on an actual `BgImageContent::Raster` value. `first_gradient_stop_count` is refactored to delegate to the same helper, so the gradient arms are covered by the other tests. All arms reached, no dead code in any single test.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: crates/fulgur/src/convert/style/background.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (mitsuru) [reply crates/fulgur/src/convert/style/background.rs]: Follow-up: the approach in `153945d` was further refined in `0d3dbe2` based on a subsequent coderabbit finding. The final fix extracts `first_gradient_stop_count_in_drawables(drawables: &crate::drawables::Drawables) -> Option<usize>` and the raster test passes the bundle-built `&drawables` directly to it, so the `_ => None` arm is exercised on an actual `BgImageContent::Raster` value. `first_gradient_stop_count` delegates to the same helper. All match arms covered, no dead code.

---
_Generated by [Claude Code](https://claude.ai/code)_
